src/sprightly.ts

```diff
@@ -1,52 +1,43 @@
-import { readFileSync } from "fs"
+import { readFileSync, existsSync } from "fs"
 import path from "path"
 
 interface Data {
   [key: string]: string | number | Array<string | number | Data> | Data
 }
 
-class KeyNotFoundError extends Error {
-  readonly key: string
-  readonly src: string
-  constructor(key: string, src: string) {
-    const message = `Key "${key}" not found in "data" at ${src}`
-    super(message)
-    this.key = key
-    this.src = src
-  }
-}
-
-class PathNotFoundError extends Error {
-  readonly component: string
-  readonly src: string
-  constructor(component: string, src: string) {
-    const message = `Component "${component}" was not found in ${src}`
-    super(message)
-    this.component = component
-    this.src = src
-  }
-}
-
 interface Options {
   keyFallback?: string
   throwOnKeyNotfound?: boolean
   cache?: boolean
 }
 
-function parse(file: string, data: Data, options: Options) {
+function parse(file: string, entryPoint: string, data: Data, options: Options) {
   const parsedFile = file.replaceAll(
     /\{\{(>?)(.*?)\}\}/g,
     (_, isComponent: string, reference: string) => {
       reference = reference.trim()
       if (isComponent) {
-        try {
-          return sprightly(reference, data, options)
-        } catch (e) {
-          throw new PathNotFoundError(reference, "")
+        // if last segement of path is a file
+        // we go back one step to get the directory of the file
+        if (path.extname(entryPoint)) {
```

Comment:
I don't think this check is necessary. In fact, you can remove these whole 3 lines and replace them with the code I suggested in the next block.

Reply:
I found out what the issue was, I was modifying `entryPoint`, which at the first iteration gave out a directory. But the other iterations got a directory at the last segment of the path (instead of a file name).

Consequences of side effects, crazy eh?